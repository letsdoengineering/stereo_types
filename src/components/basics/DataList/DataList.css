.data_list {
    display: flex;
    flex-direction: column;
}

.data_list_item-key {
    display: inline-block;
    margin-inline-start: 0;
}

.data_list_item-value {
    display: inline-block;
    margin-inline-start: 0;
}
