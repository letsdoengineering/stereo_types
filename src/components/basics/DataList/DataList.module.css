.data-list {
    display: flex;
    flex-direction: column;
}

.item-key {
    display: inline-block;
    margin-inline-start: 0;
}

.item-value {
    display: inline-block;
    margin-inline-start: 0;
}
