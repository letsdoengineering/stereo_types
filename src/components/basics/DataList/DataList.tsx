import React from 'react'
import getClassNames from '../../../utils/get-class-names'
import * as dataListStyles from './DataList.module.css'

export type DataListProps = React.AllHTMLAttributes<HTMLDListElement> & {
  /** optional: class-names */
  className?: string
  /** children always of type ReactNode */
  children: React.ReactNode
}

/** DataList: renders a DataList tag to wrap DataListValue and DataListKey pairs in  */
const DataList: React.FC<DataListProps> = ({ className, children }: DataListProps) => {
  const listClassNames = getClassNames({
    defaultClasses: [`${dataListStyles.dataList}`],
    className,
  })

  return <dl className={listClassNames}>{children}</dl>
}

export default DataList
