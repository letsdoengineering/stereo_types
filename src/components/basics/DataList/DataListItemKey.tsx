import React from 'react'
import getClassNames from '../../../utils/get-class-names'
import dataListStyles from './DataList.module.css'

type DataListItemKeyProps = React.AllHTMLAttributes<HTMLElement> & {
  /** optional: class-names */
  className?: string
  /** children always of type ReactNode */
  children: React.ReactNode
}

/** DataListItemKey: renders a DT tag */
const DataListItemKey: React.FC<DataListItemKeyProps> = ({
  className,
  children,
}: DataListItemKeyProps) => {
  const classNames = getClassNames({
    defaultClasses: [`${dataListStyles.itemKey}`],
    className,
  })

  return <dt className={classNames}>{children}</dt>
}

export default DataListItemKey
