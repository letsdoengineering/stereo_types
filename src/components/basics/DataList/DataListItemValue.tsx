import React from 'react'
import getClassNames from '../../../utils/get-class-names'
import './DataList.css'

type DataListItemValueProps = React.AllHTMLAttributes<HTMLElement> & {
  /** optional: class-names */
  className?: string
  /** children always of type ReactNode */
  children: React.ReactNode
}

/** DataListItemValue: renders a DD tag */
const DataListItemValue: React.FC<DataListItemValueProps> = ({
  className,
  children,
}: DataListItemValueProps) => {
  const classNames = getClassNames({
    defaultClasses: ['data_list_item-value'],
    className,
  })

  return <dd className={classNames}>{children}</dd>
}

export default DataListItemValue
