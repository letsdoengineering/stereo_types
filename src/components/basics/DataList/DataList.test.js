import * as React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { axe } from 'jest-axe'

import DataList from './DataList'
import DataListItemKey from './DataListItemKey'
import DataListItemValue from './DataListItemValue'

describe('DataList', () => {
  it('has not changed the JSX layout of the page from previous snapshot', () => {
    const tree = renderer
      .create(
        <DataList>
          <DataListItemKey>Name</DataListItemKey>
          <DataListItemValue>Tester123</DataListItemValue>
          <DataListItemKey>Age</DataListItemKey>
          <DataListItemValue>123</DataListItemValue>
        </DataList>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders a list that is accessible', async () => {
    const { container } = render(
      <DataList>
        <DataListItemKey>Name</DataListItemKey>
        <DataListItemValue>Tester123</DataListItemValue>
        <DataListItemKey>Age</DataListItemKey>
        <DataListItemValue>123</DataListItemValue>
      </DataList>
    )
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
