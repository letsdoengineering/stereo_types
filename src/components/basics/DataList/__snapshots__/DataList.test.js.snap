// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataList has not changed the JSX layout of the page from previous snapshot 1`] = `
<dl
  className="dataList"
>
  <dt
    className="itemKey"
  >
    Name
  </dt>
  <dd
    className="itemValue"
  >
    Tester123
  </dd>
  <dt
    className="itemKey"
  >
    Age
  </dt>
  <dd
    className="itemValue"
  >
    123
  </dd>
</dl>
`;
